### PR TITLE
Added certificate authentication to cli

### PR DESCRIPTION
**Pull request by z0ok**
Small change that allows request to authenticate with client certificate+key. Added example to readme. Tested successfully on my instance of passbolt.

### CODE REVIEW

File: passboltapi/__init__.py [hunk 1]
```diff
@@ -71,6 +73,10 @@ def __init__(
             raise ValueError("Missing value for SERVER in config.ini")
 
         self.server_url = self.config["PASSBOLT"]["SERVER"].rstrip("/")
+
+        if self.cert_auth:
+            self.cert_auth = (self.config["PASSBOLT"]["SERVER_CERT_AUTH_CRT"], self.config["PASSBOLT"]["SERVER_CERT_AUTH_KEY"])
```
Comment (bpshubhamdipt) [inline passboltapi/__init__.py:78]:
```suggestion
            if not (self.config["PASSBOLT"]["SERVER_CERT_AUTH_CRT"] and self.config["PASSBOLT"]["SERVER_CERT_AUTH_KEY"]):
                raise ValueError("Missing certificate and key in config.ini")
            self.cert = (self.config["PASSBOLT"]["SERVER_CERT_AUTH_CRT"], self.config["PASSBOLT"]["SERVER_CERT_AUTH_KEY"])
```

File: passboltapi/__init__.py [hunk 2]
```diff
@@ -60,6 +61,7 @@ def __init__(
         """
         self.ssl_verify = ssl_verify
         self.config = config
+        self.cert_auth = cert_auth
```
Comment (bpshubhamdipt) [inline passboltapi/__init__.py:64]:
```suggestion
        self.cert_auth = cert_auth
        self.cert = None
```

File: passboltapi/__init__.py [hunk 3]
```diff
@@ -111,7 +117,7 @@ def _import_gpg_keys(self):
         self.gpg.import_keys(open(self.config["PASSBOLT"]["USER_PRIVATE_KEY_FILE"]).read())
 
     def _login(self):
-        r = self.requests_session.post(self.server_url + LOGIN_URL, json={"gpg_auth": {"keyid": self.gpg_fingerprint}}, verify=self.ssl_verify)
+        r = self.requests_session.post(self.server_url + LOGIN_URL, json={"gpg_auth": {"keyid": self.gpg_fingerprint}}, verify=self.ssl_verify, cert=self.cert_auth)
```
Comment (bpshubhamdipt) [inline passboltapi/__init__.py:120]:
```suggestion
        r = self.requests_session.post(self.server_url + LOGIN_URL, json={"gpg_auth": {"keyid": self.gpg_fingerprint}}, verify=self.ssl_verify, cert=self.cert) # None is the default value in requests
```

**Comment by bpshubhamdipt** [comment]
First of all thanks for the support. Some minor suggestions added.

**Comment by shubhamdipt** [comment]
First of all thanks for the support. Some minor comments added.

**Comment by z0ok** [comment]
Thx for quick review! Agree with all comments, added everything.

**Comment by shubhamdipt** [comment]
0.3.8 released :)

**Pull request merged**